### PR TITLE
Adding config for ocp-qe-perfscale-ci image

### PR DESCRIPTION
**Pull request by kedark3**
Signed-off-by: Kedar Vijay Kulkarni <kkulkarni@redhat.com>

### CODE REVIEW

**Comment by kedark3** [comment]
@mffiedler @liangxia @jhou1 Please look at this starter draft.

File: ci-operator/step-registry/openshift-qe/cluster-density/openshift-qe-cluster-density-commands.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+set -o errexit
+set -o nounset
+set -o pipefail
+set -x
+
+oc config view
+oc projects
+python3 -m pip install virtualenv --user
```
Comment (liangxia) [inline ci-operator/step-registry/openshift-qe/cluster-density/openshift-qe-cluster-density-commands.sh:9]: Perhaps we can move it into the Dockerfile/image, so we do not need to repeat the install every time we test, which saves us time/resource.

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    tag: latest
+build_root:
+  image_stream_tag:
+    name: ci-tools-build-root
+    namespace: ci
+    tag: latest
+images:
+- dockerfile_path: prow/Dockerfile
+  from: base
+  to: ocp-qe-perfscale-ci
```
Comment (liangxia) [inline ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main.yaml:14]: base image and to image with the same name does not looks correct to me.

File: ci-operator/step-registry/openshift-qe/cluster-density/openshift-qe-cluster-density-commands.sh (hunk 1 above)
Comment (kedark3) [reply ci-operator/step-registry/openshift-qe/cluster-density/openshift-qe-cluster-density-commands.sh:9]: Yep. Moving it there !!

File: ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main.yaml (hunk 2 above)
Comment (kedark3) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main.yaml:14]: Do I always need build_root and base_image? it is not clear to me from the docs. https://docs.ci.openshift.org/docs/architecture/ci-operator/

Comment (kedark3) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main.yaml:9]: I think base_image not required. Build_root is.

Comment (liangxia) [reply ci-operator/config/openshift-qe/ocp-qe-perfscale-ci/openshift-qe-ocp-qe-perfscale-ci-main.yaml:8]: Yes, you are right. base_image is not required, and build_root is required if you would like to build a image there.

**Comment by liangxia** [comment]
Tried in https://github.com/openshift/release/pull/24400, and it show me,
```
STEP 1: FROM python:3.8
time="2021-12-09T02:56:07Z" level=warning msg="failed, retrying in 5s ... (1/2). Error: Error determining manifest MIME type for docker://python:3.8: Error reading manifest sha256:a874dcabc74ca202b92b826521ff79dede61caca00ceab0b65024e895baceb58 in docker.io/library/python: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit"
time="2021-12-09T02:56:57Z" level=warning msg="failed, retrying in 5s ... (2/2). Error: Error determining manifest MIME type for docker://python:3.8: Error reading manifest sha256:a874dcabc74ca202b92b826521ff79dede61caca00ceab0b65024e895baceb58 in docker.io/library/python: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit"
error: build error: error creating build container: Error initializing source docker://python:3.8: Error reading manifest 3.8 in docker.io/library/python: toomanyrequests: You have reached your pull rate limit. You may increase the limit by authenticating and upgrading: https://www.docker.com/increase-rate-limit
INFO[2021-12-09T02:57:35Z] Ran for 5m0s                                 
ERRO[2021-12-09T02:57:35Z] Some steps failed:                           
ERRO[2021-12-09T02:57:35Z] 
  * could not run steps: step ocp-qe-perfscale-ci failed: the build ocp-qe-perfscale-ci failed after 3m2s with reason DockerBuildFailed: Dockerfile build strategy has failed.
```
@kedark3 @mffiedler 
So the issue is that we failed to build the image due to rate limit. Perhaps we need to switch to quay.io ?

**Comment by mffiedler** [comment]
/lgtm

@smg247 We are ready to move forward with this one.  Tests are passing and it establishes the basic image and framework needed for QE performance and scalability tests

**Review by smg247 — APPROVED** [review]

**Pull request merged**